kimchi/src/circuits/constraints.rs

```diff
@@ -468,6 +420,7 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         lookup_tables: Vec<LookupTable<F>>,
         fr_sponge_params: ArithmeticSpongeParams<F>,
         public: usize,
+        precompute: Option<Arc<DomainConstantEvaluations<F>>>,
```

Comment:
`precompute` -> `precomputations`?  Also worth adding in the documentation how this can be used by the caller (provide the precomputation or set it to None).

Now that I'm looking at this I find it a bit weird as well, the precomputation is based on the domain size and at this point in time we don't know the domain size... What if you pass a precomputation that's unrelated to the domain size we're using here?

At the very least it should perform a check that the domain are the same and return an error (or here `None` until https://github.com/o1-labs/proof-systems/pull/4907 gets in)

Reply:
oh and can you update the example in the `README.md`?

Reply:
From a user perspective it would be nice to have two functions:

* create
* create_with_shared_precomputations

where create calls create_with_shared_precomputations with precomputations set to `None`. Like that the example in the README doesn't have to change

Reply:
ok updated so the `create` calls `create_with_shared_precomputations` how shall i add this new fn to the Readme?

Reply:
I think it's better to use the `create` function in the README :o which it already does